src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/AccessRightService.java

```diff
@@ -96,6 +96,30 @@ public AccessRightService(LocalGroupRepository localGroupRepository, GlobalConfF
         this.endpointService = endpointService;
     }
 
+    public List<AccessRightHolderDto> getAccessRightHoldersByClient(ClientId clientId)
+            throws ClientNotFoundException {
+        ClientType clientType = clientRepository.getClient(clientId);
+        if (clientType == null) {
+            throw new ClientNotFoundException("Client " + clientId.toShortString() + " not found");
+        }
+        // Filter just acls that are set to base endpoints so they are on service code level
+        List<AccessRightType> serviceCodeLevelAcls = clientType.getAcl().stream()
+                .filter(acl -> acl.getEndpoint().isBaseEndpoint())
+                .collect(Collectors.toList());
+        List<AccessRightType> distinctAccessRightTypes = distinctAccessRightTypeByXroadId(serviceCodeLevelAcls);
+        return mapAccessRightsToAccessRightHolders(clientType, distinctAccessRightTypes);
+    }
+
+    private List<AccessRightType> distinctAccessRightTypeByXroadId(List<AccessRightType> acls) {
```

Comment:
It _is_ quite clear when you read the code, and has a descriptive name, but it _might_ make sense to add a tiny comment to explain what it does?

Reply:
Added short description